Migrate dag js

bbovenzi opened the pull request:
Somehow we missed one file in #14115 of migrating all js from html files to their own files. It may have been blocked on other outstanding PRs that touched the UI.

This PR fixes that for `dag.html`. Gantt, Tree, and Task Instance views also use some functions from dag.js, so that is now being imported a better way.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

bbovenzi committed:
initial pass at migrating dag js

- blocked on changes to graphjs

bbovenzi committed:
Merge branch 'master' into migrate-dag-js

bbovenzi committed:
remove duplicate head_meta

bbovenzi committed:
extra linting in dag.js

bbovenzi committed:
resolve subdag and log redirect

ryanahamilton reviewed (approved):
Looks great, thanks @bbovenzi  👏

ryanahamilton reviewed (changes requested):
Looks like some related test failures

bbovenzi committed:
wrap log redirect with if statement

show_external_log_redirect may be undefined. Wrapping <meta> tag in the if statement that was in javascript previously
